tests/test_freqtradebot.py

```diff
@@ -885,14 +909,24 @@ def test_execute_entry(mocker, default_conf_usdt, fee, limit_order,
     assert trade.open_rate_requested == 10
 
     # In case of custom entry price not float type
+    mocker.patch.multiple(
+        'freqtrade.exchange.Exchange',
+        name=exchange_name,
+        get_maintenance_ratio_and_amt=MagicMock(return_value=[0.01, 0.01])
+    )
```

Comment:
@xmatthias I'm getting the same sort of problem here with the correct values not being returned even though their mocked. In my code it looks like `get_maintenance_ratio_and_amt` is returning `[0.005, 0]` but it should return `['0.01', '0.01']`

![](https://i.imgur.com/Wfmka1e.png)

Reply:
I've only looked at the binance case for now ... but you mock the method in "Exchange" - but the instantiated class is "binance".

This is not a problem UNLESS the child-class implements this method (which binance does in this case) - so this is not mocked, but returns "some" default result.

In this case, you might be better off by using monkeypatch instead of patching (basically you assign the MagicMock to the object directly)

for example using `freqtrade.exchange.get_maintenance_ratio_and_amt=MagicMock(return_value=(0.01, 0.01))` (not as part of a patch ...).

Reply:
Thank you, that get's me closer

Reply:
Actually nvm, it's still the same